ndcube/tests/test_ndcube.py

```diff
@@ -910,3 +912,15 @@ def test_axis_world_coords_without_input(test_input, expected):
     for i in range(len(all_coords)):
         np.testing.assert_allclose(all_coords[i].value, expected[i].value)
         assert all_coords[i].unit == expected[i].unit
+
+
+@pytest.mark.parametrize("test_input,expected", [
```

Comment:
You can put in more than one expected output by making ```expected``` a ```tuple``` and then unpacking that tuple in the test.  The test should be testing the same thing(s) in the same way.  You can test multiple things in one test.

Reply:
Yes I know. Do you want I regroup types test ?

Reply:
I think it's fine to have something like:
```python
def test_():
    expected_dimensions, expected_type, ... = *expected
    assert expected_dimensions == test_input.dimensions
    assert isinstance(test_input, expected_type)
    ...
```

Reply:
I must add and other test for the metadata.

Reply:
It's normally good to have your tests do less.  But in this case, I think you could combine the two tests into one.  So in the same test you can check the dimensions, the type and the meta of the returned ```NDCubeSequence```, and the type and meta of all the sub-cubes.  The sub-cube metas should all be ```None```.

Also, one other piece of advice.  Execute the ```explode_along_axis``` method inside the test, rather than in the parameterization.  That was if it fails you will get a helpful traceback.  If it fails in the parameterization, it could be hard to understand what's gone wrong.  This just requires adding an extra line to the start of the test doing
```python
output = test_input[0].explode_along_axis(test_input[1])
```
where ```test_input```  in the parameterization is now a ```tuple```, (```NDCube```, ```int```), contain the ```NDCube``` upon which ``` explode_along_axis``` is performed, and the axis along which it should be exploded.

Reply:
Ah, I didn't realise that the sub-cube meta was an empty ```OrderedDict```.  I misunderstood your message earlier.

Regarding, the dimensions of the ```NDCubeSequence``` and sub-```NDCube```s, I didn't mean they should have the same value.  They should be checked separately within the test.  That means they need to have separate ```expected``` inputs.  If you feel it's easier to break it out into multiple tests though, go ahead.

Reply:
I clearly understand why you want to regroup tests but I didn't find a better than seperate `NDCubeSequence` and `sub-NDCube`. Like that, we can have a better understanding about which test is about which types. But I think that will be interesting to modify the quantity of list for the `sub-NDCube.dimensions` to a tuple of quantity as for `NDCubeSequence`.

Reply:
Just to clarify, you're saying ```NDCube.dimensions``` should give ```(<Quantity x pix>, <Quantity x pix>, <Quantity x pix>)``` instead of what it current gives which is ```<Quantity [x, x, x] pix>```?

Reply:
Yes, I think this should be better, as we try to have the same functions for each class.

Reply:
I would be open to considering that.  Having a Quantity of multiple numbers is obviously handier.  But you can't do that for ```NDCubeSequence``` as if there's a common axis, the length of the cubes along that axis many not be the same length.  On the other hand, you could  turn a tuple of single value quantities into a single quantity of multiple values by doing ```u.Quantity((0*u.pix, 1*u.pix), unit=u.pix)```.  Although it's probably not very fast.

Like I said, I'm open to it.  I'll think about it a little more. :)

Reply:
@Cadair could we have your thoughts on this proposed change to the ```NDCube.dimensions``` API?  The proposed change is that the output will become a tuple of length-one Quantities, rather than a Quantity of length>1.  You can see @BaptistePellorceAstro's original arguments above.  I am open to it for a couple of reasons:
* It is more consistent with the output of ```ndarray.shape``` and other such attributes;
* It makes ```NDCube.dimensions``` consistent with ```NDCubeSequence.dimensions```.  The latter is a tuple of Quantities so if the sub-cubes' along the common axis are of differing lengths, that can be expressed by the common axis Quantity in the tuple having length > 1 which gives the length of each cube.
* It can be easily converted to a single Quantity by doing ```>>> Quantity(my_cube.dimensions, unit="pix")```

This may however, have repercussion for APE 14?  If so, perhaps there are arguments to escalate this change there too?